docs/source/format/CDataInterface.rst

```diff
@@ -0,0 +1,773 @@
+| ``tDm``         | duration [milliseconds]                           |            |
++-----------------+---------------------------------------------------+------------+
+| ``tDu``         | duration [microseconds]                           |            |
++-----------------+---------------------------------------------------+------------+
+| ``tDn``         | duration [nanoseconds]                            |            |
++-----------------+---------------------------------------------------+------------+
+| ``tiM``         | interval [months]                                 |            |
++-----------------+---------------------------------------------------+------------+
+| ``tiD``         | interval [days, time]                             |            |
++-----------------+---------------------------------------------------+------------+
+
+Dictionary-encoded types do not have a specific format string.  Instead, the
+format string of the base array represents the dictionary index type, and the
+value type can be read from the dependent dictionary array (see below
+"Dictionary-encoded arrays").
+
+Nested types have multiple-character format strings starting with ``+``.  The
+names and types of child fields are read from the child arrays.
+
++------------------------+---------------------------------------------------+------------+
+| Format string          | Arrow data type                                   | Notes      |
++========================+===================================================+============+
+| ``+l``                 | list                                              |            |
++------------------------+---------------------------------------------------+------------+
+| ``+L``                 | large list                                        |            |
++------------------------+---------------------------------------------------+------------+
+| ``+w:123``             | fixed-sized list [123 items]                      |            |
++------------------------+---------------------------------------------------+------------+
+| ``+s``                 | struct                                            |            |
++------------------------+---------------------------------------------------+------------+
+| ``+m``                 | map                                               | \(2)       |
++------------------------+---------------------------------------------------+------------+
+| ``+ud:I,J,...``        | dense union with type ids I,J...                  |            |
++------------------------+---------------------------------------------------+------------+
+| ``+us:I,J,...``        | sparse union with type ids I,J...                 |            |
++------------------------+---------------------------------------------------+------------+
+
+Notes:
+
+(1)
+   The timezone string is appended as-is after the colon character ``:``, without
+   any quotes.
```

Comment:
Do you want to mention that the colon and time zone may be omitted, indicating tz-naive? Perhaps also mention for clarify that the time zone name should be known to the Olsen database